google/resource_compute_snapshot.go

```diff
@@ -1,220 +1,529 @@
+		if oldConvenience == "" && newConvenience != "" {
+			return diff.ForceNew("source_disk_encryption_key_raw")
+		} else {
+			return diff.ForceNew("source_disk_encryption_key.0.raw_key")
+		}
+	}
+
+	// convenience no longer used
+	if oldConvenience != "" && newConvenience == "" {
+		if newNewField == "" {
+			// convenience is being nulled, and the new field is empty as well
+			// we've stopped using the field altogether
+			return diff.ForceNew("source_disk_encryption_key_raw")
+		} else if oldConvenience != newNewField {
+			// convenience is being nulled, and the new field has a new value
+			// so we ForceNew on either field
+			return diff.ForceNew("source_disk_encryption_key_raw")
+		} else {
+			// If we reach it here, we're using the same value in the new field as we had in the convenience field
+		}
+	}
+
+	// new no longer used
+	if oldNewField != "" && newNewField == "" {
+		if newConvenience == "" {
+			// new field is being nulled, and the convenience field is empty as well
+			// we've stopped using the field altogether
+			return diff.ForceNew("source_disk_encryption_key.0.raw_key")
+		} else if newConvenience != oldNewField {
+			// new is being nulled, and the convenience field has a new value
+			// so we ForceNew on either field
+			return diff.ForceNew("source_disk_encryption_key.0.raw_key")
+		} else {
+			// If we reach it here, we're using the same value in the convenience field as we had in the new field
+		}
+	}
+
+	return nil
+}
+
 func resourceComputeSnapshot() *schema.Resource {
 	return &schema.Resource{
 		Create: resourceComputeSnapshotCreate,
 		Read:   resourceComputeSnapshotRead,
-		Delete: resourceComputeSnapshotDelete,
 		Update: resourceComputeSnapshotUpdate,
+		Delete: resourceComputeSnapshotDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceComputeSnapshotImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(300 * time.Second),
+			Update: schema.DefaultTimeout(300 * time.Second),
+			Delete: schema.DefaultTimeout(300 * time.Second),
+		},
+		CustomizeDiff: customdiff.All(
+			customDiffComputeSnapshotSnapshotEncryptionKeys,
+			customDiffComputeSnapshotSourceDiskEncryptionKeys,
+		),
 
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
-
-			"zone": &schema.Schema{
+			"source_disk": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"description": {
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
 				ForceNew: true,
 			},
-
-			"snapshot_encryption_key_raw": &schema.Schema{
-				Type:      schema.TypeString,
-				Optional:  true,
-				ForceNew:  true,
-				Sensitive: true,
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
```

Comment:
Optional/Computed, maybe?

Reply:
The field as-is isn't `Computed`, so let's keep it `Optional` only.